### PR TITLE
User node fix

### PR DESCRIPTION
**Pull request by Dmdv**
## Fixes
https://github.com/0chain/0chain/issues/1346

## Tests
Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/0chain/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

## Associated PRs (Link as appropriate):
- blobber:
- gosdk:
- system_test:
- zboxcli:
- zwalletcli:
- Other: ...

### CODE REVIEW

File: code/go/0chain.net/smartcontract/zcnsc/burn.go [hunk 1]
```diff
@@ -3,14 +3,13 @@ package zcnsc
...
-	"0chain.net/chaincore/currency"
-
-	"0chain.net/core/util"
-
 	cstate "0chain.net/chaincore/chain/state"
+	"0chain.net/chaincore/currency"
 	"0chain.net/chaincore/state"
 	"0chain.net/chaincore/transaction"
 	"0chain.net/core/common"
+	. "0chain.net/core/logging"
```
Comment (peterlimg) [inline code/go/0chain.net/smartcontract/zcnsc/burn.go]: Avoid using Dot import, please use `logging.Logger.` below instead.

File: code/go/0chain.net/smartcontract/zcnsc/burn.go [hunk 2]
```diff
@@ -53,6 +51,7 @@ func (zcn *ZCNSmartContract) Burn(
 			info,
 		)
 		err = common.NewError(code, msg)
+		Logger.Error(msg, zap.Error(err))
```
Comment (peterlimg) [inline code/go/0chain.net/smartcontract/zcnsc/burn.go]: Use `logging.Logger.Error`

File: code/go/0chain.net/smartcontract/zcnsc/burn.go [hunk 3]
```diff
@@ -61,37 +60,30 @@ func (zcn *ZCNSmartContract) Burn(
 	if err != nil {
 		msg := fmt.Sprintf("payload decode error: %v, %s", err, info)
 		err = common.NewError(code, msg)
+		Logger.Error(msg, zap.Error(err))
```
Comment (peterlimg) [inline code/go/0chain.net/smartcontract/zcnsc/burn.go]: As above

**Review by peterlimg — CHANGES_REQUESTED** [review]

File: code/go/0chain.net/smartcontract/zcnsc/burn.go [hunk 4]
```diff
@@ -61,37 +60,30 @@ func (zcn *ZCNSmartContract) Burn(
...
-					info,
-				),
-			)
-			return
-		}
-	case util.ErrValueNotPresent:
-		err = common.NewError(code, "user node is nil "+info)
-		return
-	default:
+	if err != nil {
```
Comment (dabasov) [inline code/go/0chain.net/smartcontract/zcnsc/burn.go:75]: not clear what has changed here

File: code/go/0chain.net/smartcontract/zcnsc/mint.go [hunk 5]
```diff
@@ -54,27 +55,18 @@ func (zcn *ZCNSmartContract) Mint(trans *transaction.Transaction, inputData []by
...
-					info,
-				),
-			)
-			return
-		}
-	case util.ErrValueNotPresent:
-		err = common.NewError(code, "user node is nil "+info)
-		return
-	default:
+	if err != nil {
```
Comment (dabasov) [inline code/go/0chain.net/smartcontract/zcnsc/mint.go:58]: why isn't it  possible to mint tokens for client that doesn't have record in the MPT yet?

**Review by dabasov — CHANGES_REQUESTED** [review]

File: code/go/0chain.net/smartcontract/zcnsc/burn.go [hunk 6]
```diff
@@ -37,14 +36,13 @@ func (zcn *ZCNSmartContract) Burn(
...
+		Logger.Error(msg, zap.Error(err))
 		return "", common.NewError(code, msg)
 	}
 
 	// check burn amount
-	transSAS, err := currency.ParseZCN(float64(trans.Value))
-	if err != nil {
-		return "", err
-	}
+	transSAS := currency.Coin(trans.Value)
```
Comment (ssardana08) [inline code/go/0chain.net/smartcontract/zcnsc/burn.go:44]: use currency.Int64ToCoin()

**Review by ssardana08 — CHANGES_REQUESTED** [review]

File: code/go/0chain.net/smartcontract/zcnsc/burn.go (hunk 6 above)
Comment (peterlimg) [reply code/go/0chain.net/smartcontract/zcnsc/burn.go:44]: We will change the txn.Value to uint64 in your currency fix PR, so should be fine leave it as it is.

File: code/go/0chain.net/smartcontract/zcnsc/burn.go (hunk 4 above)
Comment (peterlimg) [reply code/go/0chain.net/smartcontract/zcnsc/burn.go:75]: I think this does not change any logic, it just a minor refactor to return early when err is not nil.

File: code/go/0chain.net/smartcontract/zcnsc/mint.go (hunk 5 above)
Comment (peterlimg) [reply code/go/0chain.net/smartcontract/zcnsc/mint.go:58]: @dabasov , the GetUserNode function will return an empty UserNode if it got`ErrValueNotPresent` from MPT. So it should fine.

**Review by peterlimg — APPROVED** [review]
lgtm

**Review by peterlimg — APPROVED** [review]

**Review by dabasov — APPROVED** [review]
lgtm

**Review by ssardana08 — APPROVED** [review]

**Pull request merged**